src/server/server.go

```diff
@@ -24,8 +24,14 @@ const (
 	STATS_PATH  = "/stats"
 	DEBUG_PATH  = "/debug"
 	PPROF_MEM_PATH  = "/pprof/mem"
+	PPROF_CPU_START_PATH = "/pprof/cpu-start"
+	PPROF_CPU_STOP_PATH = "/pprof/cpu-stop" 
 )
 
+// temporary file storing cpu profiled data
```

Comment:
should these be protected by a lock?

Reply:
Yes, I think `cpuTempFile` should be protected by a lock, and this would not result in any loss in the performance of interest. In extreme cases, without a lock, it is possible for resource contention to occur - threads 1 & 2 enter `PprofCpuStart` at the same time (both passing the `nil` check) and create two temp files; then, thread 1 modifies `cpuTempFile` and `StartCPUProfile()` smoothly, but thread 2 then modifies it, too, and fails to `StartCPUProfile()`, in which case the value stored in `cpuTempFile` will never be valid, and the file to which the data are actually written (by thread 1) cannot be tracked. So, I added a mutex lock to avoid this in the latest update.